### ISSUE TITLE
Trends in loss training

### ISSUE THREAD

**Issue by aaronhsueh0506** [Rikorose/DeepFilterNet#69]
Hi Rikorose,

I'm now in the process of transferring your PyTorch code to Tensorflow/Keras and I'm running into some issues.
The loss factor for maskloss is 0, dfalphaloss is 1000, and spectralloss is 20000.
But I think dfalphaloss is not multiplied by 1000 in your code, the loss looks like dfalphaloss +spectrumloss * 20000.

And my training phase in Keras, dfalphaloss dropped from 0.085 to 0.06 in the first epoch, and then did not continue the downward trend.
But spectrum loss is slowly decreasing. Is this situation correctly?
I also try sisdr loss, but the effect also not good.

Another thing, when I compared the process file with your code. My .wav file looks like deepfilter(stage 2) doesn't work, and your wav is obviously processed under 5kHz. 

By the way, when I check the code in 'LocalSnrTarget', the 'ws' and 'ws_ns' are not same, so when compute the local snr in speech/noise use the different frames? (speech for 1 and noise for 3)
And I think the lsnr layer in Encoder doesn't use to compute loss?

Do you have any suggestion? 
Thanks,

**Comment by Rikorose** [comment]
Hi Aaron,

the df alpha loss should also decrease. What alpha values does your network predict? If all alphas are 0, then stage2 DF will not enhance the lower frequencies. `ws_ns` is just for averaging the energy over a slightly larger window for SNR computation. There is an SNR loss for the lsnr prediction.

**Comment by Rikorose** [comment]
It should also work without any alpha loss. I am seeing alpha values from 0 - 1 (mean ~0.2) when using the alpha loss. Without a alpha loss, alpha should be more or less consistent around 0.5.
LR schedule as described in the paper. But you should also get decent results with a constant lr of 1e-4 and adam optimizer.
